README.md

```diff
@@ -1,2 +1,102 @@
+
+Within a given object there are 3 divisions based on assumed changes in road network congestion, these are `rural`, `suburban` and `urban`. We deliberately avoid a one size fits all defintion of these categorizations as they may differ from system to system. See the [conflation](https://github.com/OpenStreetMapSpeeds/conflation) repo for more information about where the data comes from and how it is used to populate these divisions.
+
+Within each density/populational division we further classify by road types which should be somewhat familiar to those familiar with OSM. There are really only two types of information we use to do this classification, the functional road class (FRC or `highway` tag in OSM) and the form of way (FoW may come from many different tag values in OSM). The `highway` tag values that we assign speeds for are in descending importance: `motorway`, `trunk`, `primary`, `secondary`, `tertiary`, `unclassified`, `residential` and `service`.
+
+We'll start by describing those values who only rely on their form of way, ie `driveway`, `alley`, `parking_aisle` and `drive-through`. These are all tagged in OSM as `highway=service` and then `service=*` where `*` is one of `driveway`, `alley`, `parking_aisle` or `drive-through`. Because they have only one value for their `highway` tag they only apply to that 1 functional road class (the lowest) and so they have only a single value for speed.
+
+We also differentiate roundabouts which are tagged as `junction=roundabout` on OSM ways. Since they may occur with any `highway` tag from above we have an array for their speeds where the first speed in the array corresponds to the `motorway` speed and the last speed in the array corresponds to the `service` speed.
+
+For the `highway=*_link` tags we have two classifications. Those that appear with any kind of signage (eg. `destination=*`) which we call `link_exiting` or those without signage which we call `link_turning`. Note that we only support an array of speeds for the 5 most important `highway=*_link` tags ie. `motorway_link`, `trunk_link`, `primary_link`, `secondary_link` and `tertiary_link`.
+
+Finally we have all other ways which do not have a designated or implied form of way. For these we simply supply a speed per `highway` tag as mentioned above, where the first speed is for `highway=motorway` and the last speed is for `highway=service`.
+
+The schema, since it uses arrays does not allow for the omission of values however one could use `0` or `null` to signal no data for a particular categorization.
```

Comment:
guess we need to update valhalla then soon?:)

```suggestion
The schema, since it uses arrays, does not allow for the omission of values however one could use `0` or `null` to signal no data for a particular categorization.
```

Reply:
yep, i liked your suggestions in the other thread so i thought id just add it to the schema as you suggested